Source/Streamstone.Tests/Scenarios/Tracking_entity_changes.cs

```diff
@@ -211,19 +212,28 @@ public void When_inserting_entity_with_unconditional_Etag_and_entity_already_exi
                     "Insert of entity with Etag=* does not behave like InsertOrReplace");
         }
 
-        
-     /* Rules for  operation chaining
-        ----------------------------------------------------
-            --->    |    Insert    |    Replace     | Delete    
-        ----------------------------------------------------
-         Insert     |      ERR     |     Insert     | NULL     
-        ----------------------------------------------------
-         Replace    |      ERR     |    Replace     | Delete     
-        ----------------------------------------------------
-         Delete     |    Replace   |      ERR       |  ERR      
-        ----------------------------------------------------
-         NULL       |    Insert    |      ERR       |  ERR      
-        --------------------------------------------------- */
+        /*           Rules for  operation chaining
+                     -----------------------------------------------------------------------------------
+                         --->    |    Insert    |    Replace    |    Delete   |  Upmerge   |  InsReplace
```

Comment:
Let's alias InsReplace as Upsert as it was used duiring discussion of this issue.